metadata-ingestion/setup.py

```diff
@@ -112,7 +112,7 @@ def get_long_description():
 
 sql_common = {
     # Required for all SQL sources.
-    "sqlalchemy>=1.3.24, <2",
+    "sqlalchemy>=1.4.0, <2",
```

Comment:
Do we actually support 1.4.0? We probably want a stronger lower bound

```suggestion
    "sqlalchemy>=1.4.0, <2",
```

Reply:
I noticed lint failure in ./tests/test_helpers/sqlalchemy_mypy_plugin.py when using sqlalchemy==1.4.0. Upgrading to 1.4.8 lower bound solves this. Do you have any thoughts on how to decide stronger lower bound  - maybe based on recency OR quickly try out some sqlalchemy versions in test matrix and choosing the least which works .

Reply:
I suspect our true lower bound is higher than 1.4.8, but I don't think it's worth the effort of finding it just yet

For now, let's set the lower bound to something that's a year or so old (maybe 1.4.39 from Jun 2022) and add a comment explaining that it's a temporary lower bound that we're open to loosening/tightening as requirements show up

main compat issues I anticipate are gonna be from GE, so let's see how it goes

Reply:
updated lower bound to 1.4.39 and added comment in this [commit](https://github.com/datahub-project/datahub/pull/8810/commits/70a00b39e84418e3c0917f3c7819b7944bd0a6b8).